src/app/code/community/Zendesk/Zendesk/controllers/ApiController.php

```diff
@@ -18,6 +18,15 @@
 class Zendesk_Zendesk_ApiController extends Mage_Core_Controller_Front_Action
 {
 
+    public function preDispatch() {
+        parent::preDispatch();
+        // TODO!!!!!!!: read version from config.xml
+        // $configSettings = Mage::getSingleton('Zendesk_Zendesk/config');
+        // Mage::log(json_encode($configSettings), null, 'zendesk.log');
+        $this->getResponse()->setHeader('X-Extension-Version', '3.0.0');
```

Comment:
Hello - this should really be inside `_authorise` after the token is validated, to ensure the version is not exposed to unauthorised clients, as otherwise it could be used to detect vulnerable versions.